### PR TITLE
Feature/add path new

### PR DESCRIPTION
**Pull request by brunottonurb**

### CODE REVIEW

File: src/components/Path/PathStage.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { IconSVG } from '../../.';
+
+export const PathStage = (props) => {
+  const { className, assistiveText, label, complete, current, selected, onStageClick, ...rest, } = props;
+  const sldsClasses = [
```
Comment (fspoettel) [inline src/components/Path/PathStage.js]: add line before

File: src/components/Path/PathStage.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    className,
+  ];
+
+  return (
+    <li className={cx(sldsClasses)} role="presentation" {...rest}>
+      <a
+        aria-selected={selected}
+        className="slds-tabs_path__link"
+        role="option"
+        tabIndex={selected ? 0 : -1}
```
Comment (fspoettel) [inline src/components/Path/PathStage.js]: `-1` removes keyboard access. via tabbing, is this intentional?

File: src/components/Path/Path.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+export const Path = (props) => {
+  const { children, button, className, ...rest } = props;
```
Comment (fspoettel) [inline src/components/Path/Path.js]: sort pls

File: src/components/Path/PathStage.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { IconSVG } from '../../.';
+
+export const PathStage = (props) => {
+  const { className, assistiveText, label, complete, current, selected, onStageClick, ...rest, } = props;
```
Comment (fspoettel) [inline src/components/Path/PathStage.js]: sort please

File: src/components/Path/__tests__/Path.spec.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { render, shallow } from 'enzyme';
+
+import { Path } from '../Path';
+import { PathStage } from '../PathStage';
+
+describe('<Path />', () => {
+  it('renders the correct html and classes', () => {
+    const mounted = render(
```
Comment (fspoettel) [inline src/components/Path/__tests__/Path.spec.js]: > render returns a wrapper very similar to the other renderers in enzyme, mount and shallow; however, render uses a third party HTML parsing and traversal library Cheerio. We believe that Cheerio handles parsing and traversing HTML extremely well, and duplicating this functionality ourselves would be a disservice.

why is this necessary here?

File: src/components/Path/__tests__/Path.spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      <Path button={(<p className="whatishappening">test</p>)}>
+        <PathStage label="testLabel" onStageClick={() => {}} />
+      </Path>
+    );
+    expect(mounted.find('div.slds-path-coach').length).toBe(1);
```
Comment (fspoettel) [inline src/components/Path/__tests__/Path.spec.js]: `expect(mounted.find('div.slds-path-coach').exists()).toBeTruthy()`

File: src/components/Path/__tests__/Path.spec.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    expect(mounted.find('div.slds-grid').length).toBe(1);
+    expect(mounted.find('div.slds-tabs_path').length).toBe(1);
+  });
+
+  it('renders the button', () => {
+    const mounted = render(
```
Comment (fspoettel) [inline src/components/Path/__tests__/Path.spec.js]: you could extract this in a renderFunction like this:

`const getComponent = (props ={}) => shallow(...)`

and call it

File: src/components/Path/__tests__/Path.spec.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      <Path button={(<p className="whatishappening">test</p>)}>
+        <PathStage label="testLabel" onStageClick={() => {}} />
+      </Path>
+    );
+    expect(mounted.find('p.whatishappening').length).toBe(1);
```
Comment (fspoettel) [inline src/components/Path/__tests__/Path.spec.js]: store the button in a const and check with `expect(mounted.contains(buttonConst)).toBeTruthy()`

File: src/components/Path/__tests__/PathStage.spec.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  it('applies className and rest-properties', () => {
+    const mounted = shallow(
+      <PathStage label="testLabel" onStageClick={() => {}} />
+    );
+    mounted.setProps({ className: 'foo', 'data-test': 'bar' });
+    expect(mounted.find('li.slds-tabs_path__item').hasClass('foo')).toBeTruthy();
+    expect(mounted.find('li.slds-tabs_path__item').prop('data-test')).toEqual('bar');
+  });
+
+  /*
```
Comment (fspoettel) [inline src/components/Path/__tests__/PathStage.spec.js]: ?

File: src/components/Path/__tests__/PathStage.spec.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+      <PathStage label="testLabel" onStageClick={() => {}} />
+    );
+    expect(mounted.find('li.slds-tabs_path__item').length).toBe(1);
+    expect(mounted.find('a.slds-tabs_path__link').length).toBe(1);
+    expect(mounted.find('span.slds-tabs_path__stage').length).toBe(1);
+  });
+
+  it('applies className and rest-properties', () => {
+    const mounted = shallow(
+      <PathStage label="testLabel" onStageClick={() => {}} />
```
Comment (fspoettel) [inline src/components/Path/__tests__/PathStage.spec.js]: please extract to renderFunction here too.

File: src/components/Path/__tests__/PathStage.spec.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+import { shallow, mount } from 'enzyme';
+
+import { PathStage } from '../PathStage';
+
+describe('<PathStage />', () => {
+  it('renders the correct html and classes', () => {
+    const mounted = shallow(
+      <PathStage label="testLabel" onStageClick={() => {}} />
+    );
+    expect(mounted.find('li.slds-tabs_path__item').length).toBe(1);
```
Comment (fspoettel) [inline src/components/Path/__tests__/PathStage.spec.js]: see above

**Comment by fspoettel** [comment]
@bruno please also make sure to clean up the commit history before merging

File: src/components/Path/PathStage.js (hunk 2 above)
Comment (brunottonurb) [reply src/components/Path/PathStage.js]: yes: https://www.lightningdesignsystem.com/components/path/

File: src/components/Path/__tests__/PathStage.spec.js (hunk 9 above)
Comment (brunottonurb) [reply src/components/Path/__tests__/PathStage.spec.js]: How do I test span contents?

Comment (fspoettel) [reply src/components/Path/__tests__/PathStage.spec.js]: ` expect(node.text()).toEqual('test123');` doesn't work?
